graph/src/log/mod.rs

```diff
@@ -0,0 +1,31 @@
+use std::fmt::{Display, Error, Formatter};
+
+pub enum LogCode {
```

Comment:
Shall we move this to `graph/src/util/log.rs` instead of creating another `log` module?

Reply:
We can still export `LogCode` via `graph::prelude::LogCode` to provide easier access.